tests/console/nvme_checks.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Smoke tests for NVM Express Controller.
+#
+# - Install nvme-cli which is a tool to manage NVM Express
+# - Check sys path that should be hold details
+# - Check if we have any NVMe controllers on the PCI bus
+# - Check if we have NVMe devices
+# - Check if we have namespace
+# - Send Identify Admin Command to the NVMe controller
+# - Issue a Read command both using the helper and manually
+# - Compare it to what we have with a regular dd
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base "installbasetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use scheduler 'get_test_suite_data';
+
+sub _smoke_checks {
+    my $nvm_test_data = shift;
+    assert_script_run("lspci -nn | grep -i nvm");
+    assert_script_run("test -d /sys/class/nvme");
+    assert_script_run("test -c /dev/nvme || echo 'nvm character devices not found'");
+
+    my @nvm_partitions = split(/\n/, script_output("ls -l /dev/$nvm_test_data->{nvm_disk}*"));
+    foreach (@nvm_partitions) {
+        assert_script_run("test -b $_ || echo 'nvm is not a block device'");
+    }
+}
+
+sub _exercise_nvme_commands_and_validate_output {
+    my $nvm_test_data = shift;
+    # send Identify Admin Command to the NVMe controller.
+    assert_script_run("nvme id-ctrl -H /dev/$nvm_test_data->{nvm_char_device}");
+    save_screenshot;
+    my $MODE       = '\/dev\/' . $nvm_test_data->{nvm_char_device};
+    my $SN         = $nvm_test_data->{sn};
+    my $MODEL      = $nvm_test_data->{model};
+    my $NAMESPACE  = $nvm_test_data->{namespace_count};
+    my $DISK_SIZE  = $nvm_test_data->{nvm_disk_size};
+    my $BLOCK_SIZE = $nvm_test_data->{nvm_block_size};
+    my $list_regex = qr/${MODE}\s+$SN\s+${MODEL}\s+${NAMESPACE}\s+${DISK_SIZE}\s+GB.*\d+\s+GB\s+${BLOCK_SIZE}/;
+    validate_script_output("nvme list",                                         sub { $list_regex });
+    validate_script_output("nvme list-ns /dev/nvme0 | awk -F ':' '{print $2}'", sub { /0x1/ });
+    # the details of this namespace
+    # check LBA Format fields
+    # ms = Metadata Size, 0 means metadata is not supported
+    # lbads = LBA Data Size, in terms of a power of two, so LBA data size is 2⁹ = 512 bytes
+    # rp = Relative Performance, the lower the better
+    validate_script_output("nvme id-ns /dev/nvme0n1 --namespace-id=0x1", sub { m/lbaf\s+0\s+:\s+ms:0\s+lbads:9\s+rp:0\s\(in use\)/ });
```

Comment:
wondering if this should be test data as well.

Reply:
it should. Let me fix it